### PR TITLE
update auto billing

### PR DESCRIPTION
**Pull request by stsma**
- add role when retrieving sessions
- add essential logs

### CODE REVIEW

**Review by jshprz — COMMENTED** [review]
This line of code:
> var status = await _dbContext.Database.SqlQuery<string>($"exec sp_CheckIfSupervisedSessionEntered @sessIDs={pIds}").ToListAsync();

Could utilize parameterized query to prevent SQL injection vulnerability.

Proposed solution:

> var status = await _dbContext.Database.ExecuteSqlRawAsync("exec sp_CheckIfSupervisedSessionEntered @sessIDs", new SqlParameter("@sessIDs", pIds)).ToListAsync();

**Comment by stsma** [comment]
> This line of code:
> 
> > var status = await _dbContext.Database.SqlQuery<string>($"exec sp_CheckIfSupervisedSessionEntered @sessIDs={pIds}").ToListAsync();
> 
> Could utilize parameterized query to prevent SQL injection vulnerability.
> 
> Proposed solution:
> 
> > var status = await _dbContext.Database.ExecuteSqlRawAsync("exec sp_CheckIfSupervisedSessionEntered @sessIDs", new SqlParameter("@sessIDs", pIds)).ToListAsync();

Hi @jshprz , proposed solution will return number of affected rows not a custom type(which is string return type). For the original code it will automatically convert string interpolation to parameterized query. Please see https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-8.0/whatsnew#raw-sql-queries-for-unmapped-types

File: TriggeredEmailer/Services/BillingService.cs [hunk 1]
```diff
@@ -131,26 +137,31 @@ private async Task ExecuteBilling(Dictionary<int, List<vwSession>> validSessions
 
                     //need to split by student
                     //calculate total hours and pay and then insert into the table
-                    var dsSesStu = _dbContext.SessionStudents.FromSql($"exec sp_SubmitToBilling_GroupsessByStudent @strIDs={pIds}");
+                    var dsSesStu = _dbContext.SessionStudents.FromSql($"exec sp_SubmitToBilling_GroupsessByStudent @strIDs={pIds}").ToList();
```
[the remark this answers is not in the document]
Comment (stsma) [reply TriggeredEmailer/Services/BillingService.cs]: thanks @jshprz 👍 ..

File: TriggeredEmailer/Services/BillingService.cs [hunk 2]
```diff
@@ -33,6 +33,11 @@ IFileLogger<BillingService> logger
         //reset day on Sunday
         private DateTime _sunday = DateTime.Now.AddDays(-(int)DateTime.Now.DayOfWeek);
 
+        //start of billing
+        private DateTime _lastSunday => _sunday.AddDays(-7);
+        //end of billing
+        private DateTime _lastSaturday => _lastSunday.AddMinutes(_daysInterval);
```
[the remark this answers is not in the document]
Comment (stsma) [reply TriggeredEmailer/Services/BillingService.cs:39]: will definitely include this on next PR.. thank you @jshprz 👍

**Review by jshprz — APPROVED** [review]
Approved. Well done @stsma 👏

**Pull request merged**